### PR TITLE
Fix timer and push path to Firebase

### PR DESCRIPTION
**Pull request by loicmisenta**
This pull request addresses two main issues in the **RecordScreen.kt** and **RecordViewModel.kt** files.  

1. _Timer Fix_: The timer, which was previously showing negative time in some cases, has been fixed. An error check has been added to ensure that the hours, minutes, and seconds are not negative.

2. _Firebase Path Push_: The path is now correctly pushed to Firebase. When the recording is stopped, a new itinerary is created with the necessary details (including the list of LatLng points) and added to Firebase. The list of LatLng points is then cleared for the next recording.  

_In addition, the code has been refactored for better readability and organization. Import statements have been rearranged, and comments have been added or updated for clarity. The start and end dates of the recording are now also stored._

### CODE REVIEW

File: app/src/main/java/com/example/triptracker/view/map/RecordScreen.kt [hunk 1]
```diff
@@ -245,9 +254,39 @@ fun Map(
...
+                  // TODO : get location from user but not implemented yet (default device location)
+                  val flameCount = 0L
+                  val startDate = viewModel.startDate.value
+                  val endDate = viewModel.endDate.value
+                  val pinList =
+                      emptyList<Pin>() // TODO : get pin list from user but not implemented yet
+                  val description =
+                      "description" // TODO : get description from user but not implemented yet
+
+                  val itinerary =
```
Comment (theoschiff) [inline app/src/main/java/com/example/triptracker/view/map/RecordScreen.kt:273]: This looks good, I'm just wondering if this shouldn't be added in some sort of coroutine to do it asynchronously. Through the view model for instance

File: app/src/main/java/com/example/triptracker/view/map/RecordScreen.kt [hunk 2]
```diff
@@ -408,7 +447,10 @@ fun displayTime(time: Long): String {
   val seconds = TimeUnit.MILLISECONDS.toSeconds(time) % 60
   val minutes = TimeUnit.MILLISECONDS.toMinutes(time) % 60
   val hours = TimeUnit.MILLISECONDS.toHours(time)
-
+  if (seconds < 0 || minutes < 0 || hours < 0) {
```
Comment (theoschiff) [inline app/src/main/java/com/example/triptracker/view/map/RecordScreen.kt:450]: This only checks if the value seen is negative and returns an error but I don't see how it is addressed? If after doing Start -> Pause -> Stop -> Start it displays a negative value then your code just returns an error? Please tell me if I'm missing something. I would just set it back to 0. String.format("%02d:%02d:%02d", 0, 0, 0)

**Review by theoschiff — COMMENTED** [review]
Looks good, please check my comments on the negative check and on Itinerary save in the `Map` function

**Comment by loicmisenta** [comment]
The problems are fixed !

**Review by JCHAVEROT — APPROVED** [review]
Thank you for fixing the little bugs, also the asynchronous push to the DB will be very useful later with the offline mode !

**Pull request merged**